assets/js/langRedirector.js

```diff
@@ -0,0 +1,36 @@
+// Wrap all this to not pollute global namespace
+{
+    // Generate an array of languages from page
+    const JDM_LANGS = [...document.querySelectorAll('#language-dropdown > ul > li')].map(el => el.classList[0]);
+
+
+    function redirectTo(lang) {
+        const link = document.querySelector(`#language-dropdown > ul > li.${lang} > a`);
+        if (link)
+            link.click();
+    }
+
+
+    function guessedTranslation() {
+        const langs = navigator.languages;
+        for (let i=0; i < langs.length; i++) {
+            let lang = langs[i];
+
+            if (JDM_LANGS.includes(lang))
+                return lang;
+
+            // Test 'ru' in addition to 'ru-RU'
+            lang = lang.slice(0,2);
+            if (JDM_LANGS.includes(lang))
+                return lang;
+        }
```

Comment:
This algorithm for language matching is not ideal. As written, it would not redirect to `pt-BR` if the given language string was `pt`.

Reply:
49c47e9 have to move algorithm close to ideal :ok_hand:

Reply:
While that does fix the matching heuristics, [the array given by `navigator.languages` is ordered from most preferred to least preferred](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorLanguage/languages) meaning that we should try to match index 0 by any method possible before trying to match any further indices.

Reply:
Fix: b9fae2e